docs/source/getting-started/app-architecture.rst

```diff
@@ -59,14 +59,17 @@ The last part of the DAML model is the operation to add friends, called a *choic
   :end-before: -- ADDFRIEND_END
 
 DAML contracts are *immutable* (can not be changed in place), so the only way to "update" one is to archive it and create a new instance.
-That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new one with the extra friend added to the list.
+That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new one with the extra friend added to the list. Here is a quick explanation of the code: 
 
-There is some boilerplate to set up the choice (full details in the :doc:`DAML reference </daml/reference/choices>`):
+    - The choice starts with the ``nonconsuming choice`` keyword followed by the choice name ``AddFriend``.
+    - The return type of a choice is defined next. In this case it is ``ContractId User``.
+    - After that we pass arguments for the choice with ``with`` keyword. Here this is the friend we are trying to add.
```

Comment:
We don't pass arguments here, we rather declare arguments (or to be more precise, (formal) parameters).